enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,324 @@
+      zero)
+    - Could be all nodes (i.e. no successful backups have been taken)
+  - Schedule a backup pod to a random member of the set of candidate nodes
+    - A backup pod should invoke the `cluster-backup.sh` script on the host
+      via chroot and write the backup data to the host.
+    - A backup pod should first check that the available disk space on the
+      node it is running on is a multiple of the size of the node's
+      /var/lib/etcd path. A node needs a minimum amount of available storage
+      to operate reliably and automated backup must make every effort not to
+      exhaust a node's available storage.
+  - When a backup pod terminates, record the succcess or failure to a backup
+    resource
+
+### User Stories
+
+- I want to be alerted after installation or upgrade to the availability of
+  an automated backup capability and be given an indication of the importance
+  of configuring it in a timely manner.
+
+- I want to initiate a cluster backup without requiring a root shell on a
+  control plane node so as to minimize the risk involved.
+
+- I want to schedule recurring cluster backups so that I have recent cluster
+  state to recover with in the event of disaster (i.e. losing 2 control plane
+  nodes).
+
+- I want to have failure to take cluster backups for more than a configurable
+  period to be reported to me via critical alerts.
+
+- I want to minimize the operational impact of taking regular backups.
+
+### Risks and Mitigations
+
+Spreading the spread the cost of automated backups over all control plane
+nodes may not be sufficient to avoid negatively impacting cluster performance
+on IO-constrained hosts. Ideally it would be possible to limit the IO
+overhead of taking a backup via cgroup limits, but it's not clear that this
+would be achievable without the granularity provided by cgroups v2 (not
+slated for GA until RHEL9).
+
+Automated backups relying on host storage could fill up the disks of control
+plane nodes, rendering them inoperable. This is intended to be mitigated by a
+check for available disk space that ensures backups are only initiated on
+nodes that have a multiple of the space required to backup the node's
+/var/lib/etcd path.
+
+Relying on node storage may hamper recovery if all control plane nodes are
+lost. It might make sense to document the requirement to periodically ship
+backup data off the cluster. Given that the current restore procedure
+requires at least one control plane node, though, a cluster is only
+recoverable from backup if at least one of the backup-containing control
+plane nodes is available.
+
+## Design Details
+
+### Open Questions
+
+- Under what circumstances should backup pods that do not terminate
+  successfully or do not terminate within a reasonable period of time be
+  recreated?
+
+- If the node targeted for backup does not have sufficient space, should
+  there be an attempt to take a backup on a different node? This question is
```

Comment:
Yes, this should be attempted.

Reply:
Why is this failure "special"?  Why wouldn't all failures have failover behavior like this?

Reply:
This question is highlighting the potential for conflict between how many backups a customer wants to retain and how to ensure those backups are spread across nodes so that the loss of any one node or any one backup does not compromise the ability to recover a cluster. 

@hexfusion Do you see a tension between what we think might be reasonable (e.g. 3 backups per node) vs what customers uninformed about the consequences of time-travel might want?

Reply:
I would expect no one to want time travel if they understood the implications. in which case the goal should be validation of the etcd state file on disk as a valid restore medium. backup and time travel should be break glass